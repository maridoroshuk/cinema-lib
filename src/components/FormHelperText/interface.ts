export interface IFormHelperTextProps {
  message: string;
}
