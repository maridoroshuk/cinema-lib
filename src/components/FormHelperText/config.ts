import { theme } from '@/theme/theme';

export const background = {
  success: theme.colors.yellowGreen,
  fail: theme.colors.melon,
  info: theme.colors.lemon,
};
