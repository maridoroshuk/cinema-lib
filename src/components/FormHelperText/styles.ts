import styled from 'styled-components';

export const Wrapper = styled.div`
  padding: ${({ theme }) => theme.spaces[20]}px;
  width: 100%;
  font-family: ${({ theme }) => theme.fontFamily.Poppins};
  font-size: ${({ theme }) => theme.sizes.fontSize[20]}px;
  border-radius: ${({ theme }) => theme.spaces[10]}px;
  color: ${({ theme }) => theme.colors.neutral000};
  background-color: ${({ theme }) => theme.colors.neutral300Transparent};
`;
