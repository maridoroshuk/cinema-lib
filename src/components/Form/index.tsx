import React from 'react';
import { Formik } from 'formik';

import { IFormProps } from './interface';
import { StyledForm } from './styles';

export const Form = <T extends {}>({
  children,
  initialValues,
  validate,
  onFormSubmit,
}: IFormProps<T>) => {
  return (
    <Formik
      initialValues={initialValues}
      validate={validate}
      onSubmit={onFormSubmit}
    >
      <StyledForm>{children}</StyledForm>
    </Formik>
  );
};
