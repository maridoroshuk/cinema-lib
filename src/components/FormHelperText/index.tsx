import React from 'react';

import { IFormHelperTextProps } from './interface';
import { Wrapper } from './styles';

export const FormHelperText = ({ message }: IFormHelperTextProps) => {
  return <Wrapper>{message}</Wrapper>;
};
