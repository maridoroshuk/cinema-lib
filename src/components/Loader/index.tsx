import React from 'react';
import ReactLoading from 'react-loading';
import { ILoaderProps } from './interface';
import { LoaderWrapper } from './styles';

export function Loader({ height, width, color }: ILoaderProps) {
  return (
    <LoaderWrapper>
      <ReactLoading type="spokes" color={color} height={height} width={width} />
    </LoaderWrapper>
  );
}
