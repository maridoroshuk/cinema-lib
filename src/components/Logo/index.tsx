import React from 'react';

import LogoImage from '@/assets/Logo.png';
import { getLogoSize } from '@/helpers/getLogoSize';

import { LogoStyled } from './styles';
import { ILogo } from './interface';

export const Logo = ({ size }: ILogo) => {
  const width = getLogoSize(size);
  return (
    <LogoStyled src={LogoImage} width={width} alt="company logo (Modsen)" />
  );
};
