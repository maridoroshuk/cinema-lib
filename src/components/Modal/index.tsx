import React from 'react';

import { IModal } from './interface';
import { Wrapper } from './styles';

export const Modal = ({ children }: IModal) => {
  return <Wrapper>{children}</Wrapper>;
};
