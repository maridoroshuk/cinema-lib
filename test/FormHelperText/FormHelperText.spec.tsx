import React from 'react';
import { render } from '@testing-library/react';

import { FormHelperText } from '@/components/FormHelperText';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('FormHelperText', () => {
  it('renders success helper text ', () => {
    const { container } = render(
      setupTestProvider(
        <FormHelperText message="The data was updated" type="success" />,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({
      'background-color': '#CEE3A0',
    });
  });
  it('renders info helper text ', () => {
    const { container } = render(
      setupTestProvider(<FormHelperText message="Loading" type="info" />),
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({
      'background-color': '#FDF8C3',
    });
  });
  it('renders fail helper text ', () => {
    const { container } = render(
      setupTestProvider(
        <FormHelperText message="The fetching is failed" type="fail" />,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({
      'background-color': '#F3B6A5',
    });
  });
});
