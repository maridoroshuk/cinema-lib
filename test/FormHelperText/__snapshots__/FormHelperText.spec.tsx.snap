// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHelperText renders fail helper text  1`] = `
.c0 {
  padding: 20px;
  width: 100%;
  font-family: 'Poppins',sans-serif;
  font-size: 20px;
  border-radius: 10px;
  color: #000000;
  background-color: #F3B6A5;
}

<div
  class="c0"
>
  The fetching is failed
</div>
`;

exports[`FormHelperText renders info helper text  1`] = `
.c0 {
  padding: 20px;
  width: 100%;
  font-family: 'Poppins',sans-serif;
  font-size: 20px;
  border-radius: 10px;
  color: #000000;
  background-color: #FDF8C3;
}

<div
  class="c0"
>
  Loading
</div>
`;

exports[`FormHelperText renders success helper text  1`] = `
.c0 {
  padding: 20px;
  width: 100%;
  font-family: 'Poppins',sans-serif;
  font-size: 20px;
  border-radius: 10px;
  color: #000000;
  background-color: #CEE3A0;
}

<div
  class="c0"
>
  The data was updated
</div>
`;
